lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1514,7 +1514,11 @@ def customize_vm(self, vm_obj):
                 ident.guiUnattended.autoLogonCount = self.params['customization'].get('autologoncount', 1)
 
             if 'timezone' in self.params['customization']:
-                ident.guiUnattended.timeZone = self.params['customization']['timezone']
+                # Check if timezone value is a int before proceeding.
```

Comment:
This will fail if user provides `int` values. 
```
>>> a = 1
>>> a.isdigit()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'int' object has no attribute 'isdigit'
```

you might want to add additional check like 
```python
timezone = self.params['customization'].get(timezone')
if isinstance(timezone, int):
    ident.guiUnattended.timeZone = timezone
elif isinstance(timezone, string_types) and timezone.isdigit():
    ident.guiUnattended.timeZone = int(timezone)
else:
    self.module.fail_json(msg="customization.timezone attribute should be an integer value.")
```

Reply:
@Akasurde Thanks for the input.
I have added a helper function to return int value else return error.

This can be used in 6 other places in the file. Will update once this is approved.